### ISSUE TITLE
[Feature request] extend graph to dynamic graph

### ISSUE THREAD

**Issue by zhuyuecai** [danielegrattarola/spektral#189]
The idea is to extend graph object by add a timestamp such that it becomes the GraphSnapshot object. 
Then a DynamicGraph is a sequence of graph snapshots sorted by the timestamp.

A dynamic dataset to download and load some published dynamic graph data set.
A corresponding loader class to feed the data in batches. 

This is for an experiment with ST-GCN and TGAT

**Comment by danielegrattarola** [comment]
Hi, 

this is definitely a good idea, I would love to have support for dynamic graphs in Spektral. 
For what you propose, the timestep can be added to a regular graph object by setting it as `graph.timestamp = 'whatever'`.

Depending on the kind of application, it may be interesting to see whether existing datasets can be used as-is or if we need to define a new kind of Dataset that supports many time series (if the time series is just one, then a simple dataset would work already). Also, the creation of a custom Loder might be necessary.

Are you actively working on this now? I am happy to collaborate. 

Cheers

**Comment by zhuyuecai** [comment]
Yep, I am working on this now. The original design is the same as what you proposed. 
But then when I start working on a dataset that is for dynamic graph, I find that almost all public available dynamic graph data sets are interaction graph, and the storage schema is a continuous one ( event base but not time step based). 
Moreover, if we store all graph snapshots in memory, one snapshot would have 9000 more nodes and the A matrix is huge.
So I actually need to change the DynamicGraph object from a snapshot-based storage schema to event-based. 
And the loader would generate the A matrix for multiple snapshots on the fly.  

But yes, I am actively working on getting dynamic graph supported in Spektral.

**Comment by danielegrattarola** [comment]
Cool! Looking forward to see what you create

**Comment by relativeflux** [comment]
@zhuyuecai Do you have an update on this? It would be a great feature to have. Thanks.
